src/mmda/parsers/pdfplumber_parser.py

```diff
@@ -214,11 +214,11 @@ def parse(self, input_pdf_path: str) -> Document:
                 all_row_ids.extend(
                     [i + last_row_id + 1 for i in line_ids_of_fine_tokens]
                 )
-                last_row_id = all_row_ids[-1]
+                last_row_id = all_row_ids[-1] if all_word_ids else -1
```

Comment:
One the empty pages all_word_ids is going to be empty. I am doing a check before referencing element -1, which raises an error in case of an empty list.